### PR TITLE
Implement GetIdentity in the node API

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

What does this PR do? Why is it needed?

This PR implements Node API's `GetIdentity` method according to the spec: https://ethereum.github.io/eth2.0-APIs/#/Node/getNetworkIdentity

Which issues(s) does this PR fix?

Part of #7510

Other notes for review

N/A

### CODE REVIEW

File: beacon-chain/rpc/nodev1/node.go [hunk 1]
```diff
@@ -9,14 +9,53 @@ import (
...
+	ctx, span := trace.StartSpan(ctx, "nodeV1.GetIdentity")
+	defer span.End()
+
+	peerId := ns.PeerManager.PeerID().Pretty()
+
+	serializedEnr, err := p2p.SerializeENR(ns.PeerManager.ENR())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not obtain enr")
+	}
+	enr := fmt.Sprint("enr:", serializedEnr)
```
Comment (rauljordan) [inline beacon-chain/rpc/nodev1/node.go:29]: Use a regular `+` to add strings as it is the fastest approach in Go: https://dev.to/pmalhaire/concatenate-strings-in-golang-a-quick-benchmark-4ahh

File: beacon-chain/rpc/nodev1/node.go [hunk 2]
```diff
@@ -9,14 +9,53 @@ import (
...
 func (ns *Server) GetIdentity(ctx context.Context, _ *ptypes.Empty) (*ethpb.IdentityResponse, error) {
-	return nil, errors.New("unimplemented")
+	ctx, span := trace.StartSpan(ctx, "nodeV1.GetIdentity")
+	defer span.End()
+
+	peerId := ns.PeerManager.PeerID().Pretty()
+
+	serializedEnr, err := p2p.SerializeENR(ns.PeerManager.ENR())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not obtain enr")
```
Comment (rauljordan) [inline beacon-chain/rpc/nodev1/node.go:27]: Wondering: are we handling the error code stuff somewhere else?

File: beacon-chain/rpc/nodev1/node.go [hunk 3]
```diff
@@ -9,14 +9,53 @@ import (
...
+	var p2pAddresses []string
+	for _, address := range ns.PeerManager.Host().Addrs() {
+		p2pAddresses = append(p2pAddresses, fmt.Sprint(address.String(), "/p2p/", peerId))
+	}
+
+	disc, err := ns.PeerManager.DiscoveryAddresses()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not obtain discovery address")
+	}
+	var discoveryAddresses []string
```
Comment (rauljordan) [inline beacon-chain/rpc/nodev1/node.go:40]:
```suggestion
	discoveryAddresses := make([]string, len(disc))
```
and then do not use append but rather assignment at index

File: beacon-chain/rpc/nodev1/node_test.go [hunk 4]
```diff
@@ -41,3 +56,91 @@ func TestGetHealth(t *testing.T) {
...
+				ipv6Found = true
+			}
+		}
+		assert.Equal(t, true, ipv4Found, "IPv4 discovery address not found")
+		assert.Equal(t, true, ipv6Found, "IPv6 discovery address not found")
+		assert.Equal(t, discAddr1.String(), resp.Data.DiscoveryAddresses[0])
+		assert.Equal(t, discAddr2.String(), resp.Data.DiscoveryAddresses[1])
+	})
+
+	t.Run("ENR failure", func(t *testing.T) {
```
Comment (rauljordan) [inline beacon-chain/rpc/nodev1/node_test.go:114]: Very nice to check the unhappy paths

File: beacon-chain/rpc/nodev1/node.go (hunk 2 above)
Comment (rkapka) [reply beacon-chain/rpc/nodev1/node.go:27]: This should be handled on the client-side of the gRPC call.

File: beacon-chain/rpc/nodev1/node_test.go (hunk 4 above)
Comment (rkapka) [reply beacon-chain/rpc/nodev1/node_test.go:114]: Thanks!

File: beacon-chain/rpc/nodev1/node.go (hunk 2 above)
Comment (prestonvanloon) [reply beacon-chain/rpc/nodev1/node.go:27]: If you don't specify a code on the server side, it will be a 500.

Comment (rkapka) [reply beacon-chain/rpc/nodev1/node.go:27]: Oops, this should be a `status.Error` of course.

**Review by rauljordan — DISMISSED** [review]

**Pull request merged**